### ISSUE TITLE
Bug: ERROR unknown public IP echo HTTP provider: https://ip.3322.net

### ISSUE THREAD

**Issue by eeoa** [qdm12/ddns-updater#661]
1. Is this urgent: Yes
2. DNS provider(s) you use: Cloudflare
3. Program version:
Running version latest built on 2024-02-13T10:44:47.174Z (commit 6a6b1a8)
4. What are you using to run the container: unraid docker
5. Extra information (optional)

docker command line
```sh
docker run
  -d
  --name='ddns-updater'
  --net='bridge'
  -e TZ="Asia/Shanghai"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="{credentials}"
  -e HOST_CONTAINERNAME="ddns-updater"
  -e 'PUBLICIP_FETCHERS'='http'
  -e 'PUBLICIP_HTTP_PROVIDERS'='https://ip.3322.net'
  -e 'PUBLICIP_DNS_PROVIDERS'='all'
  -e 'PUBLICIP_DNS_TIMEOUT'='3s'
  -e 'UPDATE_COOLDOWN_PERIOD'='1m'
  -e 'HTTP_TIMEOUT'='10s'
  -e 'LOG_LEVEL'='debug'
  -e 'BACKUP_DIRECTORY'='/updater/data/backup'
  -e 'BACKUP_PERIOD'='0'
  -e 'LOG_LEVEL'='info'
  -e 'LOG_CALLER'='hidden'
  -e 'ROOT_URL'='/'
  -e 'SHOUTRRR_ADDRESSES'=''
  -e 'HEALTH_SERVER_ADDRESS'='127.0.0.1:9999'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='http://[IP]:[PORT:8000]'
  -l net.unraid.docker.icon='https://raw.githubusercontent.com/DiamondPrecisionComputing/unraid-templates/master/templates/img/ddnsgopher.png'
  -p '8000:8000/tcp'
  -v '{credentials}':'/updater/data':'rw' 'qmcgaw/ddns-updater' 
```

Logs:

```log
Running version latest built on 2024-02-13T10:44:47.174Z (commit 6a6b1a8)

🔧 Need help? https://github.com/qdm12/ddns-updater/discussions/new
🐛 Bug? https://github.com/qdm12/ddns-updater/issues/new
✨ New feature? https://github.com/qdm12/ddns-updater/issues/new
☕ Discussion? https://github.com/qdm12/ddns-updater/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2024-02-28T01:57:15+08:00 INFO Settings summary:
├── HTTP client
|   └── Timeout: 10s
├── Update
|   ├── Period: 5m0s
|   └── Cooldown: 5m0s
├── Public IP fetching
|   ├── HTTP enabled: yes
|   ├── HTTP IP providers
|   |   └── https://ip.3322.net
|   ├── HTTP IPv4 providers
|   |   └── all
|   ├── HTTP IPv6 providers
|   |   └── all
|   └── DNS enabled: no
├── Resolver: use Go default resolver
├── Server
|   ├── Listening address: :8000
|   └── Root URL: /
├── Health
|   └── Server listening address: 127.0.0.1:9999
├── Paths
|   └── Data directory: /updater/data
├── Backup: disabled
└── Logger
    ├── Level: info
    └── Caller: hidden
2024-02-28T01:57:15+08:00 INFO reading JSON config from file /updater/data/config.json
2024-02-28T01:57:15+08:00 INFO Found single setting to update record
2024-02-28T01:57:15+08:00 INFO Reading history from database: domain {credentials} host {credentials} ipv4
2024-02-28T01:57:15+08:00 ERROR unknown public IP echo HTTP provider: https://ip.3322.net
2024-02-28T01:57:15+08:00 INFO Shutdown successful
```

Configuration file:

```json
{
    "settings": [
        {
          "provider": "cloudflare",
          "zone_identifier": "{credentials}",
          "domain": "{credentials}",
          "host": "{credentials}",
          "email": "{credentials}",
          "token": "{credentials}",
          "ip_version": "ipv4",
          "ttl": 60
        }
      ]
}
```

Host OS:
Unraid 6.12.4

**Comment by qdm12** [comment]
TLDR: use `url:https://ip.3322.net`

Thanks for taking the time to report it 💯 
The code was there (for years!), but the documentation was wrong! It's fixed in https://github.com/qdm12/ddns-updater/issues/661 indicating to add the prefix `url:` to use a custom url

**Comment by eeoa** [comment]
========================================
========================================
============= ddns-updater =============
========================================
=========== Made with ❤️ by ============
======= https://github.com/qdm12 =======
========================================
========================================

Running version latest built on 2024-02-13T10:44:47.174Z (commit 6a6b1a8)

🔧 Need help? https://github.com/qdm12/ddns-updater/discussions/new
🐛 Bug? https://github.com/qdm12/ddns-updater/issues/new
✨ New feature? https://github.com/qdm12/ddns-updater/issues/new
☕ Discussion? https://github.com/qdm12/ddns-updater/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2024-03-04T00:59:48+08:00 ERROR settings validation: public ip settings: HTTP IP providers: value is not one of the possible choices: url:https://ip.3322.net
2024-03-04T00:59:48+08:00 INFO Shutdown successful

**Comment by qdm12** [comment]
Fixed for good (and tested this time) with commit 877531c3d943418878acf2aead2a1e8531c041cf

Sorry about that!
